backend/routes/user.py

```diff
@@ -4,30 +4,63 @@
 import requests
 from fastapi import APIRouter, Request
 from fastapi.encoders import jsonable_encoder
-from fastapi.responses import JSONResponse
 from utils import collect_ingredients_categories
 from constants import MEAL_API_BASE_URL
 from schemas import AuthItem
 import jwt_auth
+import hashlib
 
 router = APIRouter(prefix="/api/user")
 
-
 @router.post("/login")
 @router.options("/login")
-async def login_user(login_item: AuthItem, request: Request):
-    user_query = jsonable_encoder(login_item)
+async def login_user(login_data: AuthItem, request: Request):
+    user_query = jsonable_encoder(login_data)
     user = request.app.database.collection.find_one(user_query)
     if user:
-        encoded_jwt = jwt_auth.get_encoded_jwt(user_query)
-        set_jwt = {"$set": {"token": encoded_jwt}}
-        request.app.database.collection.update_one(user_query, set_jwt)
-        return {'token': encoded_jwt, "username": user_query["username"]}
+        if jwt_auth.verify_password(user_query["password"], user["password"]):
+            encoded_jwt = jwt_auth.get_encoded_jwt(user_query)
+            set_jwt = {"$set": {"token": encoded_jwt}}
+            request.app.database.collection.update_one(user_query, set_jwt)
+            return {'token': encoded_jwt, "username": user_query["username"]}
+        else:
+            return {'message': 'Login failed. Wrong password.'}
     else:
         return {'message': 'Login failed. No such user.'}
 
 
 @router.post("/register")
 @router.options("/register")
-def login_user(register_data: AuthItem):
-    pass
+async def register_user(register_data: AuthItem, request: Request):
+    user_query = jsonable_encoder(register_data)
+    existing_user = request.app.database.collection.find_one({"username": user_query["username"]})
+    if existing_user:
+        return {"message": "User already exists"}
+    
+    hashed_password = hashlib.sha256(user_query["password"].encode()).hexdigest()
+    encoded_jwt = jwt_auth.get_encoded_jwt(user_query)
+
+    user_query["password"] = hashed_password
+    user_query["token"] = encoded_jwt
+    user_query["ingredients"] = ''
+    user_query["meals"] = ''
```

Comment:
It is better not to store jwt in DB. The idea is to validate user jwt using cryptography, but not compare it to DB data.

Reply:
that was a suggestion just for now. Later we will improve that.